### PR TITLE
Add sample downgrade script

### PR DESCRIPTION
**Pull request by mkindahl**
This commit add a sample downgrade script that allow downgrading from
2.2.0 to 2.1.1. It is basically a copy of different files that would be
needed to build a downgrade script.

### CODE REVIEW

**Review by erimatnor — COMMENTED** [review]
A couple of questions.

* How is this tested? I guess we need a downgrade test similar to update tests. Otherwise it is difficult to know that this does something useful given the size of the downgrade script.
* Is the downgrade script automatically generated like the update scripts? The commit message seems to imply it is not. Then what is the purpose of adding this sample script?


Also, a small nit in the commit message: "This commit add_s_..."

**Comment by mkindahl** [comment]
> A couple of questions.
> 
>     * How is this tested? I guess we need a downgrade test similar to update tests. Otherwise it is difficult to know that this does something useful given the size of the downgrade script.

It is tested manually:

```
set -e

export PATH="/usr/local/pgsql/12.3/bin:$PATH"
PGROOT=`mktemp -d -t 'pgsql-XXXX'`
export PGDATA=$PGROOT/data
PGLOG=$PGROOT/log

function on_exit () {
    if pg_ctl status; then
	pg_ctl stop
    fi
}

trap on_exit EXIT

mkdir -p $PGDATA $PGLOG

# Create and start a scratch server
pg_ctl initdb
pg_ctl start -l $PGLOG/postgres.log -o "-cshared_preload_libraries='timescaledb'"
createdb $USER

# Create a 2.2.0 version from this source tree and add a hypertable.
psql -q <<EOF
CREATE EXTENSION timescaledb VERSION '2.2.0';
\dx
EOF

psql -q <<EOF
DROP TABLE IF EXISTS conditions CASCADE;
DROP TABLE IF EXISTS devices CASCADE;

CREATE TABLE conditions(
    time TIMESTAMP NOT NULL,
    device_id INTEGER,
    temperature FLOAT,
    humidity FLOAT
);

SELECT * FROM create_hypertable('conditions', 'time', 'device_id', 12);

INSERT INTO conditions
SELECT time, (random()*3 + 1)::int, random()*80 - 40, random()*100
FROM generate_series(now() - INTERVAL '28 days', now(), '1 hour') AS time;

SELECT * FROM conditions LIMIT 10;
EXPLAIN SELECT * FROM conditions;
EOF

psql -q <<EOF
ALTER EXTENSION timescaledb UPDATE TO '2.1.1';
\dx
EOF

# psql -q <<EOF
# SELECT proname, lanname, prosrc, probin
#   FROM pg_proc JOIN pg_language lang ON lang.oid = prolang
#  WHERE lanname = 'c' AND probin = '$libdir/timescaledb-2.2.0.so';
# EOF

psql -q <<EOF
SELECT * FROM conditions LIMIT 10;
EXPLAIN SELECT * FROM conditions;
EOF
```

> 
>     * Is the downgrade script automatically generated like the update scripts? The commit message seems to imply it is not. Then what is the purpose of adding this sample script?

No, it is not automatically generated. This is the next step, this is just about providing a proof-of-concept that we can actually do downgrades.

> 
> 
> Also, a small nit in the commit message: This commit add_s_

Ah, yes, I always miss that.

File: sql/updates/README.md [hunk 1]
```diff
@@ -45,14 +45,13 @@ to consider.
...
-Notes on post_update.sql
-   We use a special config var (timescaledb.update_script_stage )
-to notify that dependencies have been setup and now timescaledb 
-specific queries can be enabled. This is useful if we want to, 
-for example, modify objects that need timescaledb specific syntax as
-part of the extension update).
-The scripts in post_update.sql are executed as part of the `ALTER
-EXTENSION` stmt.
+Notes on `post_update.sql`
+   We use a special config var (`timescaledb.update_script_stage`) to
```
[the remark this answers is not in the document]
Comment (mkindahl) [reply sql/updates/README.md]: Yes, but it doesn't add to the commit, so removing it.

**Comment by mkindahl** [comment]
Closing issue since it was decided to not merge this into the master branch and instead implement the automatic generation of downgrade scripts in #3310

**Pull request closed without merging**